nixos/modules/services/hardware/buffyboard.nix

```diff
@@ -0,0 +1,122 @@
+    services.buffyboard = with lib; {
+      enable = mkEnableOption "buffyboard framebuffer keyboard (on-screen keyboard)";
+      package = mkPackageOption pkgs "buffybox" { };
+
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = ''
+          Extra CLI arguments to pass to buffyboard.
+        '';
+        example = [
+          "--geometry=1920x1080@640,0"
+          "--dpi=192"
+          "--rotate=2"
+          "--verbose"
+        ];
+      };
+
+      settings = mkOption {
+        description = ''
+          Settings to include in /etc/buffyboard.conf.
+          Every option here is strictly optional:
+          Buffyboard will use its own bakend-in defaults for those options left unset.
+        '';
+        type = types.submodule {
+          freeformType = ini.type;
+
+          options.input.pointer = mkOption {
+            type = types.nullOr types.bool;
+            default = null;
+            description = ''
+              Enable or disable the use of a hardware mouse or other pointing device.
+            '';
+          };
+          options.input.touchscreen = mkOption {
+            type = types.nullOr types.bool;
+            default = null;
+            description = ''
+              Enable or disable the use of the touchscreen.
+            '';
+          };
+
+          options.theme.default = mkOption {
+            type = types.either types.str (
+              types.enum [
+                null
+                "adwaita-dark"
+                "breezy-dark"
+                "breezy-light"
+                "nord-dark"
+                "nord-light"
+                "pmos-dark"
+                "pmos-light"
+              ]
+            );
+            default = null;
+            description = ''
+              Selects the default theme on boot. Can be changed at runtime to the alternative theme.
+            '';
+          };
+          options.quirks.fbdev_force_refresh = mkOption {
+            type = types.nullOr types.bool;
+            default = null;
+            description = ''
+              If true and using the framebuffer backend, this triggers a display refresh after every draw operation.
+              This has a negative performance impact.
+            '';
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.packages = [ cfg.package ];
+    systemd.services.buffyboard = {
+      # upstream provides the service (including systemd hardening): we just configure it to start by default
+      # and override ExecStart so as to optionally pass extra arguments
+      serviceConfig.ExecStart = [
+        "" # clear default ExecStart
+        (lib.escapeShellArgs ([ (lib.getExe' cfg.package "buffyboard") ] ++ cfg.extraFlags))
+      ];
+      wantedBy = [ "getty.target" ];
+      before = [ "getty.target" ];
+    };
+
+    environment.etc."buffyboard.conf".source = configFile;
```

Comment:
Instead of a global file, we could use the `--config-override` CLI argument. Are there any downsides to that approach?

Reply:
It looks like it's kind of the same with the config override on unl0kr.

There should be no downsides, but a completely empty config file should be tested first

Reply:
i'll add an option `services.buffyboard.configfile` that defaults to the ini generated from `services.buffyboard.settings`, then just use `--config-override ${services.buffyboard.configfile}` as CLI argument.

Reply:
i've updated the patch to use `--config-override`.

with `services.buffyboard.enable = true` and everything defaulted, the module generates a config file like this:
```ini
[input]

[quirks]

[theme]
```

it's not literally an empty file, but buffyboard accepts it. i can test the empty file if you think that's important, but seems the only way we'd end up with that is if the user explicitly overrode the module for that.